src/main/java/duke/parser/Parser.java

```diff
@@ -0,0 +1,63 @@
+package duke.parser;
+
+import duke.command.AddCommand;
+import duke.command.Command;
+import duke.command.DeleteCommand;
+import duke.command.DoneCommand;
+import duke.command.ExitCommand;
+import duke.command.FindCommand;
+import duke.command.ListCommand;
+import duke.exception.DukeException;
+
+public class Parser {
+
+    /**
+     * Parses any user inputs and determines the type of command that was entered.
+     *
+     * @param fullCommand The command input by user
+     * @return The type of command entered
+     * @throws DukeException In the event that a command has an incorrect format/ empty description.
+     */
+    public static Command parse(String fullCommand) throws DukeException {
+        //Parse the string to check which command it is referring to
+        String[] details = fullCommand.trim().split("\\s+");
+
+        try {
+            switch (details[0]) {
+            case "list":
+                return new ListCommand();
+            case "find":
+                return new FindCommand(details[1]);
+            case "done":
+                return new DoneCommand(Integer.parseInt(details[1]) - 1);
```

Comment:
You may consider moving the "minus one" process to the lowest level implementation, i.e. inside `TaskList` class, such that all these details will not be revealed to upper level classes.